src/picongpu/include/particles/filter/IsHandleValid.hpp

```diff
@@ -0,0 +1,83 @@
+/* Copyright 2017 Rene Widera
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#pragma once
+
+#include "simulation_defines.hpp"
+
+namespace picongpu
+{
+namespace particles
+{
+namespace filter
+{
+
+namespace acc
+{
+
+    //! check the particle handle
+    struct IsHandleValid
+    {
+
+        /** check particle handle
+         *
+         * @tparam T_Particle PMacc::Particles, type of the particle
```

Comment:
offline discussed: the type before the comma describe the interface which is assumed to be implemented